ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/activity/PaywallActivityLauncher.kt

```diff
@@ -0,0 +1,43 @@
+package com.revenuecat.purchases.ui.revenuecatui.activity
+
+import androidx.activity.ComponentActivity
+import androidx.activity.result.ActivityResultCallback
+import androidx.activity.result.ActivityResultLauncher
+import androidx.fragment.app.Fragment
+import com.revenuecat.purchases.Offering
+
+/**
+ * Implement this interface to receive the result of the paywall activity.
+ */
+interface PaywallResultHandler : ActivityResultCallback<PaywallResult>
+
+/**
+ * Launches the paywall activity. You need to create this object during the activity's onCreate.
+ * Then launch the activity at your moment of choice
+ */
+class PaywallActivityLauncher {
+
+    private var activityResultLauncher: ActivityResultLauncher<PaywallActivityArgs>
+
+    /**
+     * Creates a new PaywallActivityLauncher from an activity.
+     */
+    constructor(activity: ComponentActivity, resultHandler: PaywallResultHandler) {
+        activityResultLauncher = activity.registerForActivityResult(PaywallContract(), resultHandler)
+    }
+
+    /**
+     * Creates a new PaywallActivityLauncher from a fragment.
+     */
+    constructor(fragment: Fragment, resultHandler: PaywallResultHandler) {
+        activityResultLauncher = fragment.registerForActivityResult(PaywallContract(), resultHandler)
+    }
+
+    /**
+     * Launch the paywall activity.
+     * @param offering The offering to be shown in the paywall. If null, the current offering will be shown.
+     */
+    fun launch(offering: Offering? = null) {
```

Comment:
should we add `@JvmOverloads` here?

Reply:
Yup, good point! Haven't been adding API tests yet so I missed that

Reply:
Done in https://github.com/RevenueCat/purchases-android/pull/1317/commits/6826ef782fd523f79d47306107c008b28d24beb6